Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -141,7 +141,8 @@ private extension ProductStore {
                                  excludedProductIDs: excludedProductIDs) { [weak self] result in
             switch result {
             case .success(let products):
-                let shouldDeleteExistingProducts = pageNumber == Default.firstPageNumber
+                //let shouldDeleteExistingProducts = pageNumber == Default.firstPageNumber
+                let shouldDeleteExistingProducts = false
```

Comment:
Regarding this - for coupon creation I had to delete the existing products when searching since in Coupons we have to combine queries with filters - so it is necessary to clear data for every new search, or we'll end up with incorrect results.

Expected:

https://user-images.githubusercontent.com/5533851/184575678-686d7514-33a8-4f21-9609-1b38e0bda04f.mp4

Without deleting existing products:

https://user-images.githubusercontent.com/5533851/184575704-378e3516-4b86-4934-87f3-460a8401c330.mp4

I expected the deletion to not affect the product list by clearing filters and query when dismissing the product selector (like in [here](https://github.com/woocommerce/woocommerce-ios/blob/3da9580fed5f6e9fe2597c8167a934ce6a6da301/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/OrderForm.swift#L285)) - but then I wasn't aware of the issue this causes to the product search list (my bad).

Could we instead add a new param `shouldDeleteExistingProducts` to the `ProductAction.searchProducts`, keep the value as `false` by default, and set it to `true` for coupons? That would be a quick fix for now.

Reply:
Sorry @itsmeichigo and @rachelmcr, I committed this line by mistake 🤦 

The changes made in this PR are made so the Order Creation module does not rely on how the products gets deleted by searches & filters.

So this line is not needed and I have just reverted it back.

@itsmeichigo regarding your suggestion, lets continue the discussion on the main issue https://github.com/woocommerce/woocommerce-ios/issues/7464